src/app/pages/instruction-viewer/instruction-viewer.component.ts

```diff
@@ -528,52 +529,58 @@ export class InstructionViewerComponent implements OnInit, OnDestroy {
         const camera: OrthographicCamera = scene.userData["camera"];
         // get its position relative to the page's viewport
         const rect = element.getBoundingClientRect();
+        const rectParent = element.parentElement!.getBoundingClientRect();
         const canvasrect = this.canvas.getBoundingClientRect();
-        rect.x -= canvasrect.x;
-        // set the viewport
-        const width = rect.right - rect.left;
-        const height = rect.bottom - rect.top;
-        const left = rect.left;
-        const bottom = this.canvas.clientHeight - rect.bottom;
-        this.renderer.setViewport(left, bottom, width, height);
-        this.renderer.setScissor(left, bottom, width, height);
+        const top = rect.y - canvasrect.y;
+        const left = rect.x - canvasrect.x;
+        console.log(left, rect.height - top, rect.width, rect.height);
+        this.renderer.setSize(rectParent.width, rectParent.height);
```

Comment:
first thing is here, the canvas size is not the canvas size, in html (yes it makes no sense)

canvas is always in pixels, but you can size it with css, but this wont be reflected during rendering

this is the reason the canvas was pixelated, and also why you got no rendering at all, because the internal canvas sizes was completly fucked up. Here i get the div that contains the canvas (you cant just check it with the canvas as it will converge to 0), and set the canvas pixels to match it

you will need to do it on every resize tho.

After this fix, the canvas is now correctly sized and the units will match